mwdb/web/src/components/ShowObject/Views/RelatedFilesTab.js

```diff
@@ -0,0 +1,220 @@
+
+import { ObjectContext } from "@mwdb-web/commons/context";
+import {
+    ObjectAction,
+    ObjectTab,
+    ConfirmationModal,
+} from "@mwdb-web/commons/ui";
+import { APIContext } from "@mwdb-web/commons/api/context";
+import { humanFileSize, downloadData } from "@mwdb-web/commons/helpers";
+import ReactModal from "react-modal";
+
+async function updateRelatedFiles(api, context) {
+    const { setObjectError, updateObjectData } = context;
+    try {
+        let response = await api.getListOfRelatedFiles(context.object.sha256);
+        updateObjectData({
+            related_files: response.data.related_files,
+        });
+    } catch (error) {
+        setObjectError(error);
+    }
+}
+
+function RelatedFileItem({ file_name, file_size, sha256 }) {
+    const api = useContext(APIContext);
+    const context = useContext(ObjectContext);
+    const [isConfirmationModalOpen, setConfirmationModalOpen] = useState(null);
+
+    return (
+        <tr>
+            <td>{file_name}</td>
+            <td>{humanFileSize(file_size)}</td>
+            <td>
+                <Link
+                    to={`/file/${context.object.sha256}/related_files`}
+                    className="nav-link"
+                    onClick={async () => {
+                        let content = await api.downloadRelatedFile(
+                            context.object.sha256,
+                            sha256
+                        );
+                        downloadData(
+                            content.data,
+                            file_name,
+                            "application/octet-stream"
+                        );
+                    }}
+                >
+                    <FontAwesomeIcon icon={faDownload} size="1x" />
+                    &nbsp;Download
+                </Link>
+            </td>
+            <td>
+                <Link
+                    to={`/file/${context.object.sha256}/related_files`}
+                    className="nav-link"
+                    onClick={() => setConfirmationModalOpen(true)}
+                >
+                    <FontAwesomeIcon icon={faTrash} size="1x" />
+                    &nbsp;Remove
+                </Link>
+                <ConfirmationModal
+                    isOpen={isConfirmationModalOpen}
+                    onRequestClose={() => {
+                        setConfirmationModalOpen(false);
+                    }}
+                    onConfirm={async () => {
+                        await api.deleteRelatedFile(
+                            context.object.sha256,
+                            sha256
+                        );
+                        updateRelatedFiles(api, context);
+                        setConfirmationModalOpen(false);
+                    }}
+                    message={`Are you sure you want to delete this related file?`}
+                    buttonStyle="btn-success"
+                    confirmText="Yes"
+                />
+            </td>
```

Comment:
I think that Download and remove can be only icons under a column named Actions. We can save space for a longer filename this way.
Even better, would be to make it modern. Something like this

<img width="790" alt="image" src="https://user-images.githubusercontent.com/20182642/207905282-c32f5d1f-366e-4a58-9144-b31297989aec.png">

Reply:
Thank for suggestions!
I moved download and remove to one column and rearranged the layout - I think it's similar to your suggestion.
It looks like this:
![image](https://user-images.githubusercontent.com/75318192/208079697-71c396a2-d7bb-402e-9c8e-7fcf83cf6d98.png)
What do you think about it?

Reply:
Looks much better! woah! :)

Reply:
Just make sure to show the filenames\size as link, like usually down in MWDB

Reply:
Thanks!

Hmmm
I was thinking about that, but I don't know where would this link redirect...
And showing them as n-action link would be misleading
Most of the links redirect to the main list of samples/config/blobs with a search query.
Currently, we don't support querying related files or querying by them. (Unless it's something useful in your opinion?)
So I wouldn't show them as links
Unless you have other idea for link destination?

Reply:
I think it will be wise to make it queryible in some way, not necessarily directly. For example, if I am a new researcher that analyzes Emotet for the first time, and I want to download IDB of emotet from one of the previous research done by a researcher from my company. Then, I'd love to find the Emotet sample on my MWDB instance that has an IDB attached to it.

Something like

`tag:"emotet" AND related.name:"*.idb"` 

and then I will get the 3 Emotet samples on my MWDB with an IDB attached to it (from hundreds of Emotet samples on MWDB.